### ISSUE TITLE
Fix mismatch between kubectl docs

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#25943]
**This is a Feature Request**

**What would you like to be added**
Work with SIG CLI to incorporate the kubectl documentation inside https://kubernetes.io/docs/

**Why is this needed**
https://kubectl.docs.kubernetes.io/references/kubectl/ has one look; https://kubernetes.io/docs/reference/kubectl/ has another.

**Comments**
I'm thinking of using a Git submodule for this, referencing a tag (or branch?) in [k-sigs/cli-experimental](https://github.com/kubernetes-sigs/cli-experimental)

Current documentation is built from https://github.com/kubernetes-sigs/cli-experimental/tree/master/docs AIUI

/sig cli

**Comment by sftim** [comment]
BTW, the HTML content in https://kubectl.docs.kubernetes.io/ is excluded from web crawling.

**Comment by sftim** [comment]
&hellip;and there is also https://github.com/kubernetes/kubectl/tree/master/docs/book

**Comment by kbhawkey** [comment]
I agree that this should be investigated.
There are 3 sites that include kubectl reference docs.
I am in favor of publishing one reference set to avoid duplication (and time and effort).
The reference should be auto-generated and maintained so that the content is current.
The current published kubernetes.io kubectl command reference:
https://deploy-preview-25969--kubernetes-io-master-staging.netlify.app/docs/reference/generated/kubectl/kubectl-commands

**Comment by Aut0R3V** [comment]
I'd like to work on this but I'm a bit unsure of what is required here. @kbhawkey & @sftim could any of you please elaborate?

**Comment by kbhawkey** [comment]
Hi @Aut0R3V .
To start, you could read the kubectl documentation sets mentioned in this issue.I need to read how SIG-Cli builds their
documentation, https://github.com/kubernetes-sigs/cli-experimental/tree/master/docs.

**Comment by Aut0R3V** [comment]
Sure thanks!

**Comment by sftim** [comment]
@irvifa this is the issue (I couldn't find it earlier) that tracks the work to bring https://kubernetes.io/docs/reference/kubectl/ and https://kubectl.docs.kubernetes.io/references/kubectl/ into harmony.

I think it's a big enough piece of work that it'd be worth having a _brief_, synchronous discussion involving interested parties, before reaching out to SIG CLI.

**Comment by sftim** [comment]
/remove-lifecycle stale
/lifecycle frozen
(it's important)

**Comment by sftim** [comment]
Also relevant to #28742

**Comment by sftim** [comment]
_This_ issue is relevant to #30577

**Comment by sftim** [comment]
Also see https://github.com/kubernetes-sigs/reference-docs/issues/83

**Comment by Gauravpadam** [comment]
Is this issue still relevant? I'd like to help

**Comment by a-mccarthy** [comment]
@sftim and @sreeram-venkitesh for the move over, should folks working on these issues also try to update content to match the site styling? is there any concerns about content duplication?

**Comment by sftim** [comment]
The styling mostly comes over automatically; for example, compare https://kubectl.docs.kubernetes.io/guides/introduction/kubectl/ to https://k8s.io/docs/reference/kubectl/introduction/

We didn't do any special work there to match styles

**Comment by a-mccarthy** [comment]
@sftim i'm just concerned that there may be some small differences, even if the style mostly converts and it should be something we highlight when moving the pages. Or something reviewers should be made aware of.

This is also a lot of content moving over. Are the pages staying on the kubectl site as well?

**Comment by sftim** [comment]
Mmm, I (now) think these issues aren't really good first issues specifically because some tweaks _are_ needed.

**Comment by sftim** [comment]
We intend to close the experimental kubectl docs site.

:thought_balloon: @a-mccarthy how about we get people to migrate them with `draft: true` in the front matter, and then we track a separate issue, per draft page, to clean up the drafts. Does that work?

**Comment by a-mccarthy** [comment]
> 💭 @a-mccarthy how about we get people to migrate them with draft: true in the front matter, and then we track a separate issue, per draft page, to clean up the drafts. Does that work?

I guess that depends on how fast you want to take down the experimental site. If you just want the pages moved, then we can do the migration and then take the experimental site down. Is the priority to take down the experimental site or integrate the content? My concern with doing the drafts first is (as always) finding someone to do the clean up work. and then having a pile of draft content that users can't see. 

By my count there are about 20 pages being moved over across the issues (does that sound right?) 

To start, i think we should make it clear in the migration issues that there is a bit more work that goes into moving the pages over and that migration is just the first step.

**Comment by sftim** [comment]
> To start, i think we should make it clear in the migration issues that there is a bit more work that goes into moving the pages over and that migration is just the first step.

:+1:

**Comment by lmktfy** [comment]
@ashnehete, would you like to work on this? If so, we can use this as an umbrella issue and then file specific sub-issues to track individual items of work that need doing.

I am willing to provide technical advice on the changes (if needed), and **all** SIG Docs reviewers for English are willing to help out with feedback, approver nudges, etc.

**Comment by ashnehete** [comment]
Yes, I would really like to help with this project. I can also coordinate between SIG CLI and SIG Docs for this issue.

**Comment by lmktfy** [comment]
I suggest we start by filing sub issues for each of:
- redirect https://kubectl.docs.kubernetes.io/references/kubectl/ to https://kubernetes.io/docs/reference/kubectl/generated/
- deprecate https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands

How does that sound, folks?

**Comment by soltysh** [comment]
> I suggest we start by filing sub issues for each of:
> 
>     * redirect https://kubectl.docs.kubernetes.io/references/kubectl/ to https://kubernetes.io/docs/reference/kubectl/generated/
> 
>     * deprecate https://kubernetes.io/docs/reference/generated/kubectl/kubectl-commands
> 
> 
> How does that sound, folks?

I'm supportive. For the deprecated page, what's the exact plan? Put a redirect in place, or put a note about deprecation and a link to where people should look at?

**Comment by lmktfy** [comment]
> For the deprecated page, what's the exact plan? Put a redirect in place, or put a note about deprecation and a link to where people should look at?


@ashnehete if you open an issue about that deprecation – or anyone else can – then we'll have the ideal place to hold that discussion.

**Comment by lmktfy** [comment]
What still needs doing / what is the most obvious blocker for closing this issue?

**Comment by iamankushpandit** [comment]
I can help with any anything remaining. Please let me know if you need help.
